### ISSUE TITLE
Docker engine fails to start

### ISSUE THREAD

**Issue by OgeHub** [docker/for-win#13549]
### Description

After installing Docker, the engine has been failing to start. The error message screen shot is attached below.
![docker](https://github.com/docker/for-win/assets/75125594/b2d23494-4834-40b5-84f0-53b7077893be)

### Reproduce

1. Open Docker desktop
2. Engine fails to start, shows error message

### Expected behavior

Docker engine should start.

### docker version

```bash
Client:
Cloud integration:    v1.0.31
Version:  23.0.5
API version:  1.42
Go version:  go1.19.8
Git commit:  bc4487a
Built:  Wed Apr 26 16:20:14 2023
OS/Arch:  windows/amd64
Context:  default
```

### docker info

```bash
Client:
Context:    default
Debug Mode: false
Plugins:
buildx: Docker Buildx (Docker Inc.)
Version:  v0.10.4
Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
compose: Docker Compose (Docker Inc.)
Version:  v2.17.3
Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
dev: Docker Dev Environments (Docker Inc.)
Version:  v0.1.0
Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
extension: Manages Docker extensions (Docker Inc.)
Version:  v0.2.19
Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
init: Creates Docker-related starter files for your project (Docker Inc.)
Version:  v0.1.0-beta.4
Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.) 
Version:  0.6.0
Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
scan: Docker Scan (Docker Inc.)
Version:  v0.26.0
Path:     C:\Program Files\Docker\cli-plugins\docker-scan.exe
scout: Command line tool for Docker Scout (Docker Inc.)
Version:  v0.10.0
Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe
Server:
ERROR: error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/info": open //./pipe/docker_engine: The system cannot find the file specified.                     errors pretty printing info
```

### Diagnostics ID

EFC3-459F-9035-E2276C83DF32/20230615213849

**Comment by lorenrh** [comment]
We have released Docker desktop 4.21.1, containing a fix for this issue, please see https://docs.docker.com/desktop/release-notes/

**Comment by 0nly3rrors** [comment]
I get the same message in Docker Desktop 4.21.1 (114176) build
